File: swea/D3/10570.제곱 펠린드롬 수/10570.cpp
#include <iostream>
#include <string>
#include <cmath>

using namespace std;

int T; 

bool pal(int num){
  string s = to_string(num);
  string r(s.rbegin(), s.rend());
  return s == r;
}

int main(){
  cin >> T; 
  for(int t = 1; t <= T; t++){
    int a, b;
    cin >> a >> b;

    int cnt = 0;
    for(int i = (int)sqrt(a-1) + 1; i <= (int)sqrt(b); i++){
      if(pal(i) && pal(i * i)) cnt ++;
    }

    cout << '#' << t << ' ' << cnt << endl;
  }
}

File: swea/D3/10570.제곱 펠린드롬 수/10570.py
def pal(num):
    s_num = str(num)
    return s_num == s_num[::-1]

T = int(input())
for tc in range(1, T+1):
    a, b = map(int, input().split())

    cnt = 0
    for i in range(int((a - 1) ** 0.5 + 1), int(b ** 0.5) + 1):
        if pal(i) and pal(i * i):
                cnt += 1
    
    print(f'#{tc} {cnt}')